### PR TITLE
[THRIFT-4771] add nodejs THeaderProtocol support

### PR DESCRIPTION
**Pull request by dmaii**
For nodejs

This implements a subset of THeaderProtocol.
- [x] Supports TCompact and TBinaryProtocols framed/unframed for backward compatibility.
- [x] Supports buffered and framed transports

Does not support:
- Transforms
- HTTP
- Automatic detection of framed vs unframed transport. This is something I would like to address sometime in the near future in a separate PR

- [x] Did you squash your changes to a single commit?

- [x] Do you need an [Apache Jira](https://issues.apache.org/jira/projects/THRIFT/issues/) ticket?
 
- [ ] Is this change worthy of a release note? 

- [x] Does this change require a build?

### CODE REVIEW

File: test/tests.json [hunk 1]
```diff
@@ -11,8 +11,7 @@
       ],
       "protocols": [
         "binary:multi",
-        "compact:multic",
-        "header"
+        "compact:multic"
```
Comment (jeking3) [inline test/tests.json]: I thought it might have been missing... :)

**Comment by jeking3** [comment]
```
===============================================================================
*** Following 2 failures were unexpected ***:
If it is introduced by you, please fix it before submitting the code.
===============================================================================
server-client:          protocol:         transport:               result:
nodejs-hs               header            framed-ip                failure(1)
nodejs-hs               header            buffered-ip              failure(1)
```
If the haskell client is disabled against other servers for header protocol, we can disable it here as well (hopefully someone will fix it someday).

**Comment by dmaii** [comment]
Hey, I know what the issue is and I'm currently working on a fix for it. It might take a few days, we can close/re-open this PR when it's fixed.

**Comment by jeking3** [comment]
Nah, leave it open, it's fine.

File: lib/nodejs/lib/thrift/header_transport.js [hunk 2]
```diff
@@ -292,16 +292,18 @@ THeaderTransport.prototype.writeHeaders = function() {
...
+    for (var i = 0; i < headerKeys.length; i++) {
+      var key = headerKeys[i];
+      var value = headers[key];
+
+      varintWriter.writeString(key);
+      varintWriter.writeString(value);
+    }
   }
-  var headerSizeWithoutPadding = varintWriter.getOutCount();
+ var headerSizeWithoutPadding = varintWriter.getOutCount();
```
Comment (jeking3) [inline lib/nodejs/lib/thrift/header_transport.js]: indent is off, but I'm not going to stop the PR because of it.

**Comment by yevdev-fiscalnote** [comment]
👍

**Pull request merged**